lib/remoteparticipant.js

```diff
@@ -150,6 +202,13 @@ class RemoteParticipant extends Participant {
  * @event RemoteParticipant#trackSubscriptionFailed
  */
 
+/**
+ * A {@link RemoteTrack} was unpublished by the {@link RemoteParticipant}.
+ * @event RemoteParticipant#trackUnpublished
+ * @param {RemoteTrackPublication} publication - The {@link RemoteTrackPublication}
+ *   which represents the unpublished {@link RemoteTrack}
+ */
+
```

Comment:
Cool—so "trackPublished" is a new event on RemoteParticipant that precedes "trackSubscribed" and "trackUnsubscribed" will precede "trackUnpublished" (if "trackUnpublished" ever fires). It should be the case that the following hold:

```js
remoteParticipant.on('trackPublished', publication => {
  assert.equal(remoteParticipant.trackPublications.get(publication.trackSid), publication);
  assert.equal(publication.track, null);
  assert(!publication.isSubscribed);
  publication.on('subscribed', track => {
    assert.equal(remoteParticipant.tracks.get(track.sid), track);
    assert.equal(publication.track, track);
    assert(publication.isSubscribed);
    assert(track.isSubscribed);
  });
  publication.on('unsubscribed', track => {
    assert(!remoteParticipant.tracks.has(track.sid));
    assert.equal(publication.track, null);
    assert(!publication.isSubscribed);
    assert(!track.isSubscribed);
  });
});

remoteParticipant.on('trackUnpublished', publication => {
  assert(!remoteParticipant.trackPublications.has(publication.trackSid));
  assert.equal(publication.track, null);
  assert(!publication.isSubscribed);
});

remoteParticipant.on('trackSubscribed', (track, publication) => {
  assert.equal(remoteParticipant.trackPublications.get(publication.trackSid), publication);
  assert.equal(remoteParticipant.tracks.get(track.sid), track);
  assert.equal(publication.track, track);
  assert.equal(publication.trackSid, track.sid);
  assert(publication.isSubscribed);
  assert(track.isSubscribed);
});

remoteParticipant.on('trackUnsubscribed', (track, publication) => {
  assert.equal(remoteParticipant.trackPublications.get(publication.trackSid), publication);
  assert(!remoteParticipant.tracks.has(track.sid));
  assert.equal(publication.track, null);
  assert.equal(publication.trackSid, track.sid);
  assert(!publication.isSubscribed);
  assert(!track.isSubscribed);
});
```

There is probably more, but I haven't read on to the tests yet...